### PR TITLE
Use mediawiki.api.messages to load messages

### PR DESCRIPTION
**Pull request by WMDE-Fisch**
Bug: T143104

### CODE REVIEW

File: DeepCat.js [hunk 1]
```diff
@@ -394,8 +394,10 @@
 	}
 
 	function substituteTitle( input ) {
-		loadMessages( 'searchresults-title' ).done( function() {
-			$( document ).prop( 'title', mw.msg( 'searchresults-title', input ) );
+		$.when( mw.loader.using(["mediawiki.api.messages", "mediawiki.jqueryMsg"]), $.ready ).done( function() {
```
Comment (manicki) [inline DeepCat.js]: single quotes?

Comment (WMDE-Fisch) [inline DeepCat.js]: thats the problem with c&p :-)

**Comment by manicki** [comment]
This looks like a good change. I am just wondering why is this particular function waiting for message to load, and why not have this whole pile of code first load messages and then do other things?

Or does it sound like a wrong here to do here?

**Comment by WMDE-Fisch** [comment]
Is now included in #69. To your question: Loading everything upfront would lead to an API request each time the Gadget loads even if not used. To avoid this I think the best place to load the message and wait for it would be the method where it is used.

**Comment by thiemowmde** [comment]
This looks like a really good change. Why did you closed this?

**Comment by manicki** [comment]
@WMDE-Fisch oh, indeed. I keep forgetting this thing is only a gadget, and it loads different. There was no need to ask then, thanks!

@thiemowmde I was also wondering but I guess @WMDE-Fisch ended up in some rebase hell and decided to have it only included in https://github.com/wmde/DeepCat-Gadget/pull/69

**Comment by WMDE-Fisch** [comment]
Yeah included in #69 with the restructure of the whole thingy :-)

**Pull request closed without merging**